sdk/azcore/request.go

```diff
@@ -66,24 +66,36 @@ func (ov opValues) get(value interface{}) bool {
 	return ok
 }
 
+// stripQueryParams splits a string into the root and the query parameters
+// if there are no query parameters, path and a blank string are returned
+func stripQueryParams(path string) (string, string) {
+	if strings.Contains(path, "?") {
+		splitPath := strings.Split(path, "?")
+		return splitPath[0], splitPath[1]
+	}
+	return path, ""
+}
+
 // JoinPaths concatenates multiple URL path segments into one path,
-// inserting path separation characters as required.
-func JoinPaths(paths ...string) string {
+// inserting path separation characters as required. JoinPaths will preserve
+// query parameters in the root path
+func JoinPaths(root string, paths ...string) string {
 	if len(paths) == 0 {
-		return ""
-	}
-	path := paths[0]
-	for i := 1; i < len(paths); i++ {
-		if path[len(path)-1] == '/' && paths[i][0] == '/' {
-			// strip off trailing '/' to avoid doubling up
-			path = path[:len(path)-1]
-		} else if path[len(path)-1] != '/' && paths[i][0] != '/' {
-			// add a trailing '/'
-			path = path + "/"
-		}
-		path += paths[i]
+		return root
+	}
+
+	root, qps := stripQueryParams(root)
+
+	for i := 0; i < len(paths); i++ {
+		root = strings.TrimRight(root, "/")
+		paths[i] = strings.TrimLeft(paths[i], "/")
+		root += "/" + paths[i]
+	}
+
+	if qps != "" {
```

Comment:
Looking at the code coverage, there's no coverage for the containing query params parts.

Reply:
Added